File: DotNetPoly/Fields/FieldCollection.cs
using System.Linq;

namespace DotNetPoly.Fields
{
    class FieldCollection : System.Collections.Generic.Dictionary<string, Field>
    {
        public Field this[int i]
        {
            get
            {
                return Values.ElementAt(i);
            }
        }
    }

    public class HouseFieldCollection : System.Collections.Generic.Dictionary<string, HouseField>
    {
        public Field this[int i]
        {
            get
            {
                return Values.ElementAt(i);
            }
        }
    }
}

File: DotNetPoly/Fields/FieldCollection.vb
Imports System.Linq

Namespace Fields
    Friend Class FieldCollection
        Inherits System.Collections.Generic.Dictionary(Of String, Fields.BaseField)

        Overloads ReadOnly Property Item(i As Integer) As Fields.BaseField
            Get
                Return Values.ElementAt(i)
            End Get
        End Property
    End Class

End Namespace
